docs/packaging-applications/build-servers/github-actions.md

```diff
@@ -32,112 +32,229 @@ For example:
 ```
 
-```yml MacOS Runner
-name: listdeployments
+### Packaging artifacts
+To package your artifacts for deployment, configure your build to use the `OctopusDeploy/install-octopus-cli-action` developed by Octopus Deploy by adding the following step (previous steps excluded for brevity)
 
-env:
-  serverURL: your_server_name
+```yaml
+    - name: Install Octopus CLI
+      uses: OctopusDeploy/install-octopus-cli-action@v1.1.6
+      with:
+        version: latest
+```
 
-on:
-   push:
-     branches: [ main ]
-   pull_request:
-     branches: [ main ]
+Adding this Action allows your build to use the commands from the Octopus [command line interface (CLI)](https://octopus.com/docs/octopus-rest-api/octopus-cli).  Using the [pack](https://octopus.com/docs/octopus-rest-api/octopus-cli/pack), you can package your artifacts for deployment.  The following example packages the OctoPetShop components built above.
+
+```yaml
+    - name: Install Octopus CLI
+      uses: OctopusDeploy/install-octopus-cli-action@v1.1.6
+      with:
+        version: latest
+    - name: Package OctoPetShopDatabase
+      run: |
+        octo pack --id="OctoPetShop.Database" --format="Zip" --version="$PACKAGE_VERSION" --basePath="$GITHUB_WORKSPACE/artifacts/OctopusSamples.OctoPetShop.Database" --outFolder="$GITHUB_WORKSPACE/artifacts"
+    - name: Package OctoPetShopWeb
+      run: |
+        octo pack --id="OctoPetShop.Web" --format="Zip" --version="$PACKAGE_VERSION" --basePath="$GITHUB_WORKSPACE/artifacts/OctopusSamples.OctoPetShop.Web" --outFolder="$GITHUB_WORKSPACE/artifacts"
+    - name: Package OctoPetShopProductService
+      run: |
+        octo pack --id="OctoPetShop.ProductService" --format="Zip" --version="$PACKAGE_VERSION" --basePath="$GITHUB_WORKSPACE/artifacts/OctopusSamples.OctoPetShop.ProductService" --outFolder="$GITHUB_WORKSPACE/artifacts"
+    - name: Package OctoPetShopShoppingCartService
+      run: |
+        octo pack --id="OctoPetShop.ShoppingCartService" --format="Zip" --version="$PACKAGE_VERSION" --basePath="$GITHUB_WORKSPACE/artifacts/OctopusSamples.OctoPetshop.ShoppingCartService" --outFolder="$GITHUB_WORKSPACE/artifacts"
+```
 
-jobs:
+### Pushing artifacts to Octopus Server
+Once the artifacts are packaged, use the Octopus CLI Action to [push](https://octopus.com/docs/octopus-rest-api/octopus-cli/push) the packages to the Octopus Server built-in repository.  The following example pushes the packages created from the Package operation.
+
+```yaml
+    - name: Push OctoPetShop Database
+      run: |
+        octo push --package="$GITHUB_WORKSPACE/artifacts/OctoPetShop.Database.$PACKAGE_VERSION.zip" --server="${{ secrets.OCTOPUSSERVERURL }}" --apiKey="${{ secrets.OCTOPUSSERVERAPIKEY }}" --space="${{ secrets.OCTOPUSSERVER_SPACE }}"
```

Comment:
We have push action you can use instead of calling the CLI directly: https://github.com/marketplace/actions/push-package-to-octopus-deploy

https://octopus.com/blog/github-actions-for-octopus-deploy#pushing-build-artifacts-to-octopus

Reply:
Updated.